### ISSUE TITLE
tilde in include path not working in 0.5.1

### ISSUE THREAD

**Issue by j4y** [Yelp/mrjob#1308]
We have a nested `-include` in the mrjob.conf, which uses the tilde to load '~/.mrjob.conf'.  That is the mrjob.conf file includes another include file 'foo.conf', which includes '~/.mrjob.conf':

```
include:
- ~/.mrjob.conf
```

This works in version 0.4.5, but does not work in 0.4.6 to 0.5.1
```
Traceback (most recent call last):
  File "/usr/local/bin/mrjob", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/mrjob/cmd.py", line 68, in main
    commands[args[1]](args[2:])
  File "/usr/local/lib/python2.7/site-packages/mrjob/cmd.py", line 93, in _create_jf
    main(args)
  File "/usr/local/lib/python2.7/site-packages/mrjob/tools/emr/create_job_flow.py", line 25, in main
    real_main(args)
  File "/usr/local/lib/python2.7/site-packages/mrjob/tools/emr/create_cluster.py", line 206, in main
    runner = EMRJobRunner(**_runner_kwargs(args))
  File "/usr/local/lib/python2.7/site-packages/mrjob/emr.py", line 573, in __init__
    super(EMRJobRunner, self).__init__(**kwargs)
  File "/usr/local/lib/python2.7/site-packages/mrjob/runner.py", line 341, in __init__
    self._opts = self.OPTION_STORE_CLASS(self.alias, opts, conf_paths)
  File "/usr/local/lib/python2.7/site-packages/mrjob/emr.py", line 382, in __init__
    super(EMRRunnerOptionStore, self).__init__(alias, opts, conf_paths)
  File "/usr/local/lib/python2.7/site-packages/mrjob/runner.py", line 172, in __init__
    alias, conf_paths=conf_paths)
  File "/usr/local/lib/python2.7/site-packages/mrjob/conf.py", line 429, in load_opts_from_mrjob_confs
    runner_alias, path, already_loaded=already_loaded) + results
  File "/usr/local/lib/python2.7/site-packages/mrjob/conf.py", line 393, in load_opts_from_mrjob_conf
    runner_alias, include, already_loaded) + inherited
  File "/usr/local/lib/python2.7/site-packages/mrjob/conf.py", line 393, in load_opts_from_mrjob_conf
    runner_alias, include, already_loaded) + inherited
  File "/usr/local/lib/python2.7/site-packages/mrjob/conf.py", line 358, in load_opts_from_mrjob_conf
    conf = _conf_object_at_path(conf_path)
  File "/usr/local/lib/python2.7/site-packages/mrjob/conf.py", line 315, in _conf_object_at_path
    with open(conf_path) as f:
IOError: [Errno 2] No such file or directory: '/Users/jprall/Code/hadoop-jobs/~/.mrjob.conf'
```

**Comment by davidmarin** [comment]
Good catch, will try to fix this in the next version (unless someone fixes it in a pull request first!). :)

**Comment by davidmarin** [comment]
Oh right, this changed with #1179, which addressed relative include paths (see #1166).

**Comment by davidmarin** [comment]
The issue is that we're joining it to the directory of the including file before expanding (resolving `~` etc.) the path. Subtle!
